### PR TITLE
Migrate claim extended fields

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Overview

Facility claims can contain a facility_type and processing_types.
Following moving to the new taxonomy for these values, existing types
are invalid.

The following invalid values are dropped: "Other", "Boarding", "Denim
Services". All other existing values are mapped to the equivalent
taxonomy values. Extended Fields are created for the processing_types
and facility_type.

Extended fields are also created for other applicable claim fields. 

Connects #1648 

## Demo

Screen Shot 2022-02-24 at 3 17 18 PM
Screen Shot 2022-02-24 at 3 17 25 PM
Screen Shot 2022-02-24 at 3 39 33 PM

## Notes

I pulled the current `facility_production_types` and `facility_type` values from production and [created a mapping](https://docs.google.com/document/d/1qfexjxlRmdzR9_NLzG45Knc_l2-Dxhn6JEnqYyKV4eY/edit?usp=sharing) to the new system, with the assistance of the OAR team. 

Because we are filtering out / removing some invalid values, and because we need to update the actual claim object for the facility_type and processing_types fields, it wasn't possible to directly use the existing scripts for processing without a lot of edits, and it didn't make sense to rewrite them for this very specific one-off use, so I ended up copying over a lot of the code and editing where needed in the migrations file. 

As part of this migration, I checked out out an earlier version of the develop branch before the latest version of the taxonomy was applied, ran `resetdb`, and created several claims. Then I ran migrations on the current version of develop and on this branch, and confirmed that the migrations ran. 

## Testing Instructions

* Manual create old-style claims (or see notes for an optional alternate testing path).
     * Run `./scripts/server`, login, and claim two facilities. Approve the claims as an admin. Fill out the claim fields for name, native language name, parent company, number of workers, and product types. 
     * Run `./scripts/manage shell_plus`, then (use the appropriate claim ids):
```
f = FacilityClaim.objects.get(id=1)
f.facility_production_types = ['Cut and Sew / RMG', 'Boarding', 'Denim Services', 'Fusing', 'Marker Making', 'Pattern Grading', 'Sample Making', 'Trimming/Embellishment', 'Warehouse']
f.facility_type = 'Cut and Sew / RMG'
f.save()

f2 = FacilityClaim.objects.get(id=2)
f2.facility_production_types = []
f2.facility_type = 'Other'
f2.save()
```
     * Confirm in the shell that the claims match the structure of existing claims. 
* Run `./scripts/manage migrate`. 
* View the claim details and confirm that for claim 1 the new values match the expected mapping, and for claim 2 no values are present. 
* View the claimed facilities' detail pages and confirm that the extended field values match the expected taxonomy mapping. 

## Checklist

- [ ] `fixup!` commits have been squashed
- [ ] CI passes after rebase
- [ ] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Comment by TaiWilkin** [comment]
Hold off on reviewing - I've realized a key piece is missing.

**Comment by TaiWilkin** [comment]
This is ready for review now.

**Comment by jwalgran** [comment]
The build broke because we merged a PR with a new migration. Rebasing and fixing the numbering and the dependency array should be all that is needed.

**Review by jwalgran — APPROVED** [review]
I used the method of checking out 23518c68, resetting my db, creating "old style" clams, then checking out this branch and migrating. Worked perfectly for me. Nice work working through a complex process. I think copy/past over refactor is definitely the right choice for this one time data migration.

File: src/django/api/migrations/0087_migrate_claim_fields.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        else:
+            ExtendedField.objects.filter(facility_claim_id=claim.id,
+                                         field_name=extended_field).delete()
+
+
+def migrate_claim_fields(apps, schema_editor):
+    FacilityClaim = apps.get_model('api', 'FacilityClaim')
+    claims = FacilityClaim.objects.filter(status='APPROVED')
+    for claim in FacilityClaim.objects.filter(status='APPROVED').iterator():
+        if claim.id is not None:
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/migrations/0087_migrate_claim_fields.py]: It shouldn't ever be `None` - it's just a copy-paste leftover. I've removed it. Thanks for the catch!

**Comment by TaiWilkin** [comment]
Thank you for reviewing!

**Pull request merged**